devices/Bmm150/Bmm150Compensation.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Iot.Device.Magnetometer;
+using System;
+
+namespace Iot.Device.Magnetometer
+{
+    /// <summary>
+    /// Implements the Bmm150 magnetic field data (off-chip) temperature compensation functions
+    /// https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bmm150-ds001.pdf
+    /// Page 15
+    /// </summary>
+    public class Bmm150Compensation
+    {
+        /// <summary>
+        /// Returns the compensated magnetometer x axis data(micro-tesla) in float.
+        /// More details, permalink: https://github.com/BoschSensortec/BMM150-Sensor-API/blob/a20641f216057f0c54de115fe81b57368e119c01/bmm150.c#L1614
+        /// </summary>
+        /// <param name="x">axis raw value</param>
+        /// <param name="rhall">temperature compensation value (RHALL) </param>
+        /// <param name="trimData">trim registers values</param>
+        /// <returns>compensated magnetometer x axis data(micro-tesla) in float</returns>
+        public static double Compensate_x(double x, uint rhall, Bmm150TrimRegisterData trimData)
```

Comment:
Functions as variables should follow the naming guidelines, in this case CompensateX without _
Same for Y and Z

Reply:
fixed